tests/integration/components/sl-calendar-test.js

```diff
@@ -0,0 +1,773 @@
+        'January ' + currentYear,
+        'Current month and year are set correctly in the view'
+    );
+});
+
+test( 'Setting dateValuePath modifies the view correctly', function( assert ) {
+
+    const modifiedValuePath = Ember.A([
+        {
+            modifiedDatePath: new Date( 2022, 8, 17 ),
+            label: 'Today!'
+        }
+    ]);
+
+    const currentYear = 2022;
+    this.set( 'currentYear', currentYear );
+
+    const currentMonth = 9;
+    this.set( 'currentMonth', currentMonth );
+
+    this.set( 'content', modifiedValuePath );
+
+    this.render( hbs`
+        {{sl-calendar
+            action='testAction'
+            content=content
+            dateValuePath='modifiedDatePath'
+            currentYear=currentYear
+            currentMonth=currentMonth
+        }}
+    ` );
+
+    assert.equal(
+        this.$( '.sl-calendar .active' ).text().trim(),
+        17,
+        'Active day is set correctly'
+    );
+});
+
+test( 'Active day is set correctly', function( assert ) {
+
+    const currentYear = 2022;
+    this.set( 'currentYear', currentYear );
+
+    const currentMonth = 9;
+    this.set( 'currentMonth', currentMonth );
+
+    const emptyContent = Ember.A( [ {} ] );
+    this.set( 'content', emptyContent );
+
+    this.render( hbs`
+        {{sl-calendar
+            action='testAction'
+            content=content
+        }}
+    ` );
+
+    assert.equal(
+        this.$( '.sl-calendar .active' ).text().trim(),
+        '',
+        'No active day is set'
+    );
+
+    this.set( 'content', testContent );
+
+    this.render( hbs`
+        {{sl-calendar
+            action='testAction'
+            content=content
+            currentYear=currentYear
+            currentMonth=currentMonth
+        }}
+    ` );
+
+    assert.equal(
+        this.$( '.sl-calendar .active' ).text().trim(),
+        17,
+        'Active day is set correctly'
+    );
+
+    this.set( 'content', multipleTestContent );
+
+    this.render( hbs`
+        {{sl-calendar
+            action='testAction'
+            content=content
+            currentYear=currentYear
+            currentMonth=currentMonth
+        }}
+    ` );
+
+    const active = this.$( '.sl-calendar .active' );
+
+    assert.equal(
+        active.length,
+        2,
+        'There are two unique dates assigned the .active class'
+    );
+
+    assert.equal(
+        active[ 0 ].innerHTML,
+        '17',
+        'First unique date instance is correct'
+    );
+
+    assert.equal(
+        active[ 1 ].innerHTML,
+        '20',
+        'Second unique date instance is correct'
+    );
+});
+
+test( 'Setting locale to Spanish modifies the view correctly', function( assert ) {
+
+    const currentYear = 2022;
+    this.set( 'currentYear', currentYear );
+
+    const currentMonth = 9;
+    this.set( 'currentMonth', currentMonth );
+
+    this.set( 'content', testContent );
+
+    this.render( hbs`
+        {{sl-calendar
+            action='testAction'
```

Comment:
What role does this serve in this test? The component will still render as expected, especially within the context of this test, without this.